src/common.jl

```diff
@@ -1,194 +1,267 @@
-using DiffEqBase, DataStructures
+using DiffEqBase, OrdinaryDiffEq
+using StaticArrays: SVector, SizedArray
+using RecursiveArrayTools: ArrayPartition
 
-import DiffEqBase: ODEProblem, solve
+import DiffEqBase: ODEProblem, solve, ODE_DEFAULT_NORM, @.., addsteps!
 
-const warnkeywords =
-    (:save_idxs, :d_discontinuities, :unstable_check, :save_everystep,
-     :save_end, :initialize_save, :adaptive, :dt, :reltol, :dtmax,
-     :dtmin, :force_dtmin, :internalnorm, :gamma, :beta1, :beta2,
-     :qmax, :qmin, :qsteady_min, :qsteady_max, :qoldinit, :failfactor,
-     :maxiters, :isoutofdomain, :unstable_check,
-     :calck, :progress, :timeseries_steps, :tstops, :dense)
+import OrdinaryDiffEq: OrdinaryDiffEqAdaptiveAlgorithm,
+OrdinaryDiffEqConstantCache, OrdinaryDiffEqMutableCache,
+alg_order, alg_cache, initialize!, perform_step!, @unpack,
+@cache, stepsize_controller!, step_accept_controller!
+
+# TODO: check which keywords work fine
+const warnkeywords = (:save_idxs, :d_discontinuities, :unstable_check, :save_everystep,
+:save_end, :initialize_save, :adaptive, :dt, :reltol, :dtmax,
+:dtmin, :force_dtmin, :internalnorm, :gamma, :beta1, :beta2,
+:qmax, :qmin, :qsteady_min, :qsteady_max, :qoldinit, :failfactor,
+:isoutofdomain, :unstable_check,
+:calck, :progress, :timeseries_steps, :dense)
```

Comment:
Can you include some indentation; makes nicer reading the code.

Can you open an issue to  have present the TO DO indicated here?

Reply:
Thanks, added indentation and also opened the issue corresponding to this TODO